lib/api/redfish-1.0/managers.js

```diff
@@ -34,42 +55,76 @@ var getManager = controller(function(req, res) {
-        switch(type) {
-        case 'ipmi-obm-service':
-            return 'BMC';
-        default:
-            return 'ManagementController';
-        }
-    };
 
     return Promise.try(function() {
-        assert.ok(_.isArray(parts), 'invalid identifier specified: ' + identifier);
-        assert.string(parts[0], 'invalid identifier specified: ' + identifier);
-        return waterline.nodes.needByIdentifier(parts[0]);
+        if(identifier === reservedId) {
+            return waterline.nodes.find({type: 'compute'});
+        }
+        return waterline.nodes.find({id: parts[0]});
     })
-    .then(function(node) {
-        assert.ok(_.isArray(node.obmSettings), 'invalid obmSettings');
-        assert.ok(node.obmSettings.length === parseInt(parts[1]) + 1, 'invalid obmSetting');
-        options.systems = [ node.id ];
-        options.chassis = _.reduce(node.relations, function(arr,val) {
-            if(val.relationType === 'enclosedBy')  {
-                arr.push(val.targets[0]);
-            }
+    .then(function(nodes) {
+        if(_.isEmpty(nodes)) {
+            throw new Errors.NotFoundError('invalid resource');
+        }
+        if(parts[1]) {
+            _.forEach(nodes, function(node) {
+                assert.ok(_.isArray(node.obmSettings), 'invalid resource');
+                assert.ok(node.obmSettings.length > parts[1], 'invalid resource');
+            });
+        }
+
+        options.systems = _.map(nodes, 'id');
+        options.chassis = _.reduce(nodes, function(arr,val) {
+            _.forEach(val.relations, function(val) {
+                if(val.relationType === 'enclosedBy')  {
+                    arr.push(val.targets[0]);
+                }
+            });
             return arr;
         }, []);
-        options.managerType = managerType(node.obmSettings[parts[1]].service);
-        return nodeApi.getNodeCatalogSourceById(node.id, 'ipmi-mc-info');
+
+        options.managerType = identifier === reservedId ? 'ManagementController' : 'BMC';
+        if(identifier !== reservedId) {
+            return nodeApi.getNodeCatalogSourceById(nodes[0].id, 'ipmi-mc-info')
+            .then(function(mcInfo) {
+                options.mcInfo = mcInfo.data;
+            });
+        }
+    })
+    .then(function() {
+        if(identifier === reservedId) {
+            var locations = _.filter(configuration.get('httpEndpoints', []), 
+                            _.matches({routers:'northbound-api-router'}));
+            var httpLocation = _.find(locations, {httpsEnabled: false});
+            var httpsLocation = _.find(locations, {httpsEnabled: true});
+
+            var services = [];
+            if(httpLocation) {
+                services.push({ name: 'HTTP', port: httpLocation.port });
+            }
+            if(httpsLocation) {
+                services.push({ name: 'HTTPS', port: httpsLocation.port });
+            }
+            services.push({ name: 'SSDP', port: 1900 });
```

Comment:
We'll need to followup with this and add disable/enable controls per Redfish spec.